CONTRIBUTING.md

```diff
@@ -143,53 +81,110 @@ follow the following steps:
+1. After finishing up the squashing, force push your changes to your branch.
 
    ```
    git push --force-with-lease
    ```
 
+## Build and run Camunda from source
+We are currently working on [architecture streamlining](https://camunda.com/blog/2024/04/simplified-deployment-options-accelerated-getting-started-experience/) to simplify the deployment and build process. While this is in progress, the build instructions are subject to change. The most recent build instructions will always be in this document.
+
+This is a small overview of the contents of this repository:
+- `authentication` - configures authentication for Camunda 8
+- `bom` - bill of materials (BOM) for importing Zeebe dependencies
+- `build-tools` - Zeebe build tools
+- `clients` - client libraries
+- `dist` - provides the Camunda 8 distributions
+- `identity` - component within self-managed Camunda 8 responsible for authentication and authorization
+- `licenses` - the Camunda 8 licenses
+- `monitor` - Monitoring for self-managed Camunda 8
+- `operate` - Monitoring tool for monitoring and troubleshooting processes running in Zeebe
+- `parent` - Parent POM for all Zeebe projects
+- `qa` - quality assurance for Camunda 8
+- `search` - the search clients for Camunda 8 data
+- `service` - internal services for Camunda 8
+- `spring-boot-starter-sdk` - official SDK for Spring Boot
+- `tasklist` - graphical and API application to manage user tasks in Zeebe
+- `testing` - testing libraries for processes and process applications
+- `webapps-common` - shared code between the Camunda 8 web apps
+- `zeebe` - the process automation engine powering Camunda 8
+
+### Build
+
+To **quickly** build all components, run the command: `mvn clean install -DskipTests` in the root folder.
+
+> [!NOTE]
+> All Camunda core modules are built and tested with JDK 21. Most modules use language level 21, exceptions are: camunda-client-java, camunda-process-test-java, zeebe-bpmn-model, zeebe-build-tools, zeebe-client-java, zeebe-gateway-protocol zeebe-gateway-protocol-impl, zeebe-protocol, and zeebe-protocol-jackson which use language level 8.
+
+For contributions to Camunda, building quickly is typically sufficient. However, users are recommended to build the full distribution.
+
+To fully build the Camunda distribution and run all tests, run the command: `mvn clean install` in the root folder. This is slightly slower than building quickly but ensures the distribution is assembled completely. The resulting distribution can be found in the folder `dist/target`, i.e.
```

Comment:
❓ Why do we recommend running all the tests for building the Camunda distribution? Doesn't it make more sense to split it into the following?
- Building quickly for development: `mvn clean install -Dquickly`
- Building the full distribution for usage (you don't want to run all the tests): `mvn clean install -DskipChecks -DskipTests`
- Building and testing the full distribution: `mvn clean install`

Reply:
Makes sense! Thanks. Corrected